corehq/apps/hqwebapp/static/hqwebapp/js/hqModules.js

```diff
@@ -0,0 +1,24 @@
+var COMMCAREHQ_MODULES = {};
+
+function hqDefine(path, moduleObject) {
+    var parts = ['COMMCAREHQ_MODULES'].concat(path.split('/'));
+    var i;
+    var object = window;
+    for (i = 0; i < parts.length - 1; i += 1) {
+        if (typeof object[parts[i]] === 'undefined') {
+            object[parts[i]] = {};
+        }
+        object = object[parts[i]];
+    }
+    object[parts[i]] = moduleObject;
```

Comment:
Yeah great point. I think something that's coming out of this is that even though we won't get all the fancy static analysis etc. that a real module system might promise, even just doing this much gives us the opportunity to add in some runtime checks that prevent or discourage bad behavior. I'll go ahead and add this check. Thanks!